src/view/popup/popup.tsx

```diff
@@ -49,42 +52,52 @@ export const Popup = () => {
     });
 
     return (
-        <div className="p-10 w-96 flex flex-col justify-stretch gap-6">
-            {page === "landing" ? (
-                <LandingPage
-                    onRecordButtonPress={async () => {
-                        goToPage("recording");
-                        await refreshAndGetSize();
-                    }}
-                />
-            ) : page === "recording" ? (
-                <RecordingPage
-                    onStopButtonPress={async () => {
-                        await stopRecording();
-                        goToPage("results");
-                    }}
-                    bytesTransferred={bytesTransferred}
-                    emissions={emissions}
-                />
-            ) : page === "results" ? (
-                <ResultsPage
-                    onRestartButtonPress={async () => {
-                        goToPage("recording");
-                        await refreshAndGetSize();
-                    }}
-                    recordings={calculationHistory}
-                    selectedCountries={selectedCountries}
-                    addSelectedCountry={addSelectedCountry}
-                    removeSelectedCountry={removeSelectedCountry}
-                    setCountryPercentage={setCountryPercentage}
-                    userType={userType}
-                    setUserType={setUserType}
-                    error={error}
-                />
-            ) : (
-                <ErrorPage />
-            )}
-        </div>
+        <SelectedCountriesContext.Provider value={selectedCountriesContext}>
```

Comment:
@RSAMUEL12 if I were to deconstruct the useSelectedCountriesContext() hook before, do I need to pass all the deconstructed members here instead of a single object?

Reply:
yep

Reply:
is it necessary in this file then, since I just want to pass it to the provider? 🤔

I'm thinking I will deconstruct it when the methods are actually being used, e.g. in useResultsPage

what do you think?